x-pack/filebeat/input/httpjson/json_parser.go

```diff
@@ -0,0 +1,136 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package httpjson
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// parse returns array of string values from json string
+func parse(rawJSON, key string) ([]string, error) {
+	var data interface{}
+	err := json.Unmarshal([]byte(rawJSON), &data)
+	if err != nil {
+		return nil, err
+	}
+	if err != nil {
+		return nil, fmt.Errorf("error while parsing json: %w", err)
+	}
+	values, err := parseInterface(data, key)
+	if err != nil {
+		return nil, fmt.Errorf("error while parsing json: %w", err)
+	}
+	return values, nil
```

Comment:
You have duplicated error checks here, one function call and two consecutive blocks of `if err !=nil`. I also improved a bit the error message.
```suggestion
	var data interface{}
	if err := json.Unmarshal([]byte(rawJSON), &data); err != nil {
		return nil, fmt.Errorf("cannot unmarshal input data: %w", err)
	}

	values, err := parseInterface(data, key)
	if err != nil {
		return nil, fmt.Errorf("error while parsing JSON: %w", err)
	}

	return values, nil
```

Reply:
sorry about that.